marc_to_solr/lib/process_holdings_helpers.rb

```diff
@@ -154,24 +138,23 @@ def build_item(item:, field_852:, field_876:)
   end
 
   def process_866_867_868_fields(fields:, all_holdings:, holding_id:)
+    return all_holdings if all_holdings[holding_id].nil?
```

Comment:
I got the idea from your review back in February. 

I think previously we were essentially doing the same thing, but it was just sort of being hidden in the `fields.each` loop where it says `next unless all_holdings[holding_id]`. 

The idea is that there's no need to loop through the fields if `all_holdings[holding_id]` is nil, since `all_holdings[holding_id]['location_has'] << parse_textual_holdings(field, '866')` would just error out. 

Looking into it more, it seems that sometimes `all_holdings` at this point is just an empty hash, which also leads to `all_holdings[holding_id]` being nil, which seems surprising. 

I think I'm going to try to refactor the `process_holdings` method in princeton_marc to be able to reason about it more easily.

Reply:
I ended up moving this up to the `process_holdings` method, since it doesn't make sense to run the `process_866_867_868_fields` method unless `all_holdings` has the `holding_id` as a key